Common/Securities/Option/StrategyMatcher/OptionStrategyMatcherOptions.cs

```diff
@@ -0,0 +1,257 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Securities.Option.StrategyMatcher
+{
+    /// <summary>
+    /// Defines options that influence how the matcher operates.
+    /// </summary>
+    /// <remarks>
+    /// Many properties in this type are not implemented in the matcher but are provided to document
+    /// the types of things that can be added to the matcher in the future as necessary. Some of the
+    /// features contemplated in this class would require updating the various matching/filtering/slicing
+    /// functions to accept these options, or a particular property. This is the case for the enumerators
+    /// which would be used to prioritize which positions to try and match first. A great implementation
+    /// of the <see cref="IOptionPositionCollectionEnumerator"/> would be to yield positions with the
+    /// highest margin requirements first. At time of writing, the goal is to achieve a workable rev0,
+    /// and we can later improve the efficiency/optimization of the matching process.
+    /// </remarks>
+    public class OptionStrategyMatcherOptions
+    {
+        /// <summary>
+        /// The maximum amount of time spent trying to find an optimal solution.
+        /// </summary>
+        public TimeSpan MaximumDuration { get; }
+
+        /// <summary>
+        /// The maximum number of matches to evaluate for the entire portfolio.
+        /// </summary>
+        public int MaximumSolutionCount { get; }
+
+        /// <summary>
+        /// Indexed by leg index, defines the max matches to evaluate per leg.
+        /// For example, MaximumCountPerLeg[1] is the max matches to evaluate
+        /// for the second leg (index=1).
+        /// </summary>
+        public IReadOnlyList<int> MaximumCountPerLeg { get; }
```

Comment:
Shouldn't this be more of a global maximum match per strategy definition, which is the 'multiplier' right? Thinking why would a user want different `MaximumCount**PerLeg**` because even if a leg matches 10 times all of the strategy legs should match the same amount really 🤔

Reply:
This limits the total search space. Consider we have 100 unique positions, all calls. For the first leg match, we'll match all 100. This would turn into an enormous exponential number of possible matches. This parameter limits how many leg matches are evaluated. In the example above, we would limit to `MaximumCountPerLeg[0]` instead of 100. This doesn't pertain to the multiplier matches but rather unique position matches that increase the total solution space.